goss-testing/scripts/iscsi_sbps_sanity.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+#
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+# This script is to do some sanity tests for iSCSI based boot content 
+# projection.
+
+# shellcheck disable=SC2207,SC2143,SC2002
+
+Found=0
+LIO_ACT=0
+
+echo "Checking whether the node is configured with iSCSI"
+
+ISCSI=($(kubectl get nodes --selector='iscsi=sbps' -o jsonpath='{.items[*].metadata.name}'))
+
+NODE="$HOSTNAME"
+
+for i in "${ISCSI[@]}"
+do
+    if [ "${i}" == "${NODE}" ]
+    then
+        Found=1
+        echo "$NODE is configured with iSCSI"
+        break
+    fi
+done
+
+if [ "${Found}" == 0 ]
+then
+    echo "iscsi is not configured, exiting"
+    exit 1
+fi
```

Comment:
You can let `kubectl` do more work for you by asking it to select only this node:
```
ISCSI=$(kubectl get nodes --selector="iscsi=sbps,kubernetes.io/hostname=${HOSTNAME}" -o jsonpath='{.items[*].metadata.name}')
if [ -z "${ISCSI}" ]
then
    echo "iscsi is not configured, exiting"
    exit 1
else
    echo "${HOSTNAME} is configured with iSCSI"
fi
```
Also, is it considered a failure if this node isn't configured for ISCSI?  If it is okay for a node to not be configured for ISCSI, then this script should be exiting with 0 so that the caller doesn't think something failed.

Reply:
No, its not considered as failure if ISCSI is not configured.